### DISCUSSION TITLE
occupationalhazards event triggers again immediately if you pick the "Ignore" option

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#13912] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Event is triggered when you enter the crew module of the outpost. If you pick the "Ignore" option at the start, the event gets reset via GoTo name="beginning", where the triggerer_player tag gets cleared.
Since you're still in the crew module at that moment, the event immediately triggers again.

https://github.com/FakeFishGames/Barotrauma/assets/47554589/62fe8f56-03d4-48f0-b792-597cbf717b1d

To avoid this, the event should be triggered via talking to the NPC, with waitforinteraction="true".

### Reproduction steps

1. Initialize the occupationalhazards event. (triggerevent occupationalhazards)
2. Walk into the crew module to trigger the event.
3. Pick "Ignore".
4. Watch the event start again immediately.

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.4.5.0 (Blood in the Water Update, hotfix 1)

### Which operating system did you encounter this bug on?

Windows

**Comment by ubertpendragon** [comment]
Occupationalhazards event triggers by just walking into a room, and you talk to a non-existing NPC

The text then says the NPC sits into a chair

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/e188980b-8533-407a-bcda-0025a5d65764)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/bc8c20a3-2575-431e-94c4-971631c14db5)

Use this tidbit of code from `miketheidiot` event

```xml
      <TagAction criteria="itemtag:chair" tag="chair" RequiredModuleTag="CrewModule" />
      <SpawnAction npcsetidentifier="outpostnpcs1" npcidentifier="miner" targettag="hardhatman" spawnlocation="Outpost" targetmoduletags="CrewModule" />
      <NPCOperateItemAction npctag="hardhatman" targettag="chair" Operate="true" />
      <TriggerAction target1tag="player" target2tag="hardhatman" applytotarget1="triggerer_player" waitforinteraction="true"/>
```

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/b8a8cbbde2b55f56a3ed70ecba355504dee7a183 (commit in our private repo). The fix didn't make it in time for the upcoming update, but will be included in a future one.
